lib/chef/provider/windows_task.rb

```diff
@@ -100,9 +100,27 @@ class WindowsTask < Chef::Provider
 
         def load_current_resource
           @current_resource = Chef::Resource::WindowsTask.new(new_resource.name)
-          task = TaskScheduler.new
-          if task.exists?(new_resource.task_name)
+          if new_resource.task_name.split("\\").size > 1
+            *path, task_name = new_resource.task_name.split("\\")
+            path = path.join("\\")
+            begin
+              task = TaskScheduler.new(new_resource.task_name, nil, path, false)
+            rescue Exception => e
```

Comment:
Looks like this is an `ArgumentError` upstream (https://github.com/chef/win32-taskscheduler/blob/ole/lib/win32/taskscheduler.rb#L164) so no need to cast such a wide net.

Reply:
I am adding that block so that if I get folder not found error I can reset the folder to root path i.e. "\\". Otherwise just raise the error coming from the library.

Reply:
Indeed, but you can rescue `ArgumentError` instead of all exceptions it looks like. Unless that error can come from elsewhere. Slightly reduces the chances of badness :)

Reply:
I am not sure if I am missing something but I am just checking if there is folder not found error then set root folder else raise the exception in else block.

Reply:
@Vasu1105 @coderanger is saying that because in [win32-taskscheduler](https://github.com/chef/win32-taskscheduler/blob/ole/lib/win32/taskscheduler.rb#L164) we are raising `ArgumentError`, here you can `rescue ArgumentError => e` instead which is slightly safer coding behavior than rescuing Exception. This way your rescue block is only run for the type of exception you are explicitly looking for, rather than all exceptions. Rescuing all exceptions can have unintended consequences. See also https://github.com/rubocop-hq/ruby-style-guide#no-blind-rescues for one example of unintended consequences.